### ISSUE TITLE
Unable to run load-test with transaction size greater than 1MB

### ISSUE THREAD

**Issue by jovankaramac** [informalsystems/tm-load-test#51]
Hi to all,

I am trying to run load-test with 4 validators with files greater than 700KB, but it fails. How to remove the limitation? I am using the default Tendermint config file.

regards, 
Jovan

**Comment by thanethomson** [comment]
Please supply your output logs for `tm-load-test`.

**Comment by jovankaramac** [comment]
Hi Thane,

I am using KV srote, so everything is default, the config file (except for the mempool activity that is disabled), the application, everything. I am providing output1.txt from the unfinished load test with s=1000000, and output2.txt with s=700000.
[output1.txt](https://github.com/informalsystems/tm-load-test/files/4909039/output1.txt)
[output2.txt](https://github.com/informalsystems/tm-load-test/files/4909040/output2.txt)

Best regards,

**Comment by thanethomson** [comment]
Looks to me like your Tendermint RPC server endpoint's closing the connection because Tendermint is not configured to read enough data off the WebSocket connection (i.e. it's not related to `tm-load-test`).

Check the [Tendermint configuration docs](https://docs.tendermint.com/master/tendermint-core/configuration.html) on how to set up your configuration for your use case.

**Comment by jovankaramac** [comment]
Thank you Thane for your answer, I will come back with the result of the reconfiguration. Until then, stay safe. Regards,

**Comment by jovankaramac** [comment]
Thane,

I resolved the problem. I started with your suggestion and dig deep in the configuration of concurrent connections and set all VMs to handle 2M connections (I found some tutorials on the net), and this didn't resolve the issue. After that, I tried to modify all the parameters from the default config file provided from the Tendermint web (I think it was from there but I am not sure) and this still didn't solve the issue. My final hope was to go through the code of Tendermint. I the config.go file I found "tonnes" of new parameters (actually there were 4-5 parameters) that were not present or visible in the default config file. Here max_body_bytes solved the issue. I set bigger limit and the simulation finished successfully. My guess was that 1MB is the default value for transaction size (body+header) but it is strange for me that only 700kB are passing through the system. This is a bit confusing because in tm-load-test I am specifying transaction size (body+header=1MB) and they are not passing through the default setting of max_body_bytes of 1MB. 

Thank you for your suggestion, it led me to the final solution.

Regards,
